lib/MemoryModel/MemModel.cpp

```diff
@@ -189,7 +189,7 @@ bool SymbolTableInfo::computeGepOffset(const llvm::User *V, LocationSet& ls) {
         // Handling array types, skipe array handling here
         // We treat whole array as one, then we can distinguish different field of an array of struct
         // e.g. s[1].f1 is differet from s[0].f2
-        if(isa<ArrayType>(*gi))
+        if(isa<ArrayType>(gi.getIndexedType()))
```

Comment:
Ok, I did a little more testing using fix from this morning in https://github.com/jcarlson23/bridgedGEPiterator

And now that seems to conform to the 3.8 behavior pretty well now.  

That said, using API, as @dtzWill did, is clearly the direction things are moving towards so that might be preferable...  If it were up to me, I think that's what I would do.

I'll leave it up to others...  

I noticed the PTABen reference and am going to start trying to adapt my setup towards using that (I just hadn't seen it before) and might play with some alterations in the test scripts, just having a very difficult time running them at all (this is after altering the setup.sh, still seeing permissions errors, etc).